tpp-prophets/peptide_prophet.xml

```diff
@@ -18,92 +17,66 @@
             $accurate_mass
             $no_ntt
             $no_nmc
-            $use_gamma
+            $usegamma
             $use_only_expect
             $force_fit
             $allow_alt_instruments
             $maldi
             $usedecoys
-            --decoy-prefix $decoy_prefix_string
+            --decoy-prefix '$decoy_prefix_string'
 
             #if str($experiment_label).strip():
                 --experiment-label '$experiment_label'
             #end if
-
+            --database '$database'
```

Comment:
This should be an optional argument.  In a normal Galaxy workflow the input to this tool will have the correct path to the database set.  I can see why it would be useful for tests and for unusual workflows though

Reply:
I'm not sure I understand this. The XML file has a hard-coded path to the fasta database and this file is not guaranteed to stay isn't it? Assuming you share this file with me but then clean/delete your history. This file will not work anymore. Its seems this is a very implicit way of encoding a dependency a user is never aware of. Why should a user not delete the fasta file at some point if everything runs "without" this database.

Reply:
@bgruening OK but this terrible behaviour is exactly how the TPP works.  My worry is that there are consequences of this that we have not picked up with our very limited tests.